src/overlays/actors/ovl_Arms_Hook/z_arms_hook.c

```diff
@@ -23,27 +29,329 @@ const ActorInit Arms_Hook_InitVars = {
     (ActorFunc)ArmsHook_Update,
     (ActorFunc)ArmsHook_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/ArmsHook_SetupAction.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Arms_Hook/ArmsHook_Init.s")
+ColliderQuadInit quadInit = {
+    { COLTYPE_UNK10, 0x09, 0x00, 0x00, 0x08, COLSHAPE_QUAD },
+    { 0x02, { 0x00000080, 0x00, 0x01 }, { 0xFFCFFFFF, 0x00, 0x00 }, 0x05, 0x00, 0x00 },
+    { 0 },
+};
+
+Vec3f D_80865B50 = { 0.0f, 0.5f, 0.0f };                                  // unused
+Vec3f D_80865B5C = { 0.0f, 0.5f, 0.0f };                                  // unused
+Color_RGB8 D_80865B68[] = { { 0xFF, 0xFF, 0x64 }, { 0xFF, 0xFF, 0x32 } }; // unused
+
+extern Gfx D_0602B288[];
+extern Gfx D_0602AFF0[];
+Vec3f D_80865B70 = { 0.0f, 0.0f, 0.0f };
+Vec3f D_80865B7C = { 0.0f, 0.0f, 900.0f };
+Vec3f D_80865B88 = { 0.0f, 500.0f, -3000.0f };
+Vec3f D_80865B94 = { 0.0f, -500.0f, -3000.0f };
+Vec3f D_80865BA0 = { 0.0f, 500.0f, 1200.0f };
+Vec3f D_80865BAC = { 0.0f, -500.0f, 1200.0f };
+
+void ArmsHook_SetupAction(ArmsHook* this, ArmsHookActionFunc actionFunc) {
+    this->actionFunc = actionFunc;
+}
+
+void ArmsHook_Init(Actor* thisx, GlobalContext* globalCtx) {
+    ArmsHook* this = THIS;
+
```

Comment:
super small, but for small functions like this I usually dont put a line break between stack variables and code (like you did below in destroy)

Reply:
I personally think putting a line break is better, but it's not too big of a deal for small functions.

Reply:
thats fair. whichever you go with though pixel, make it consistent across all